node-js-web-server/server110.js

```diff
@@ -0,0 +1,22 @@
+const http = require('http');
+const path = require('path');
+
+http.createServer(function (request, response) {
+   console.log('request url: ', request.url);
+   
+   var filePath = './public' + request.url;
+   if (filePath == './public/') {
+      filePath = './public/index.html';
+   }
+
+   var extName = path.extname(filePath);
+}).listen(8080);
+console.log('Server is running at http://localhost:8080');
+
+// you need to make sure that the value of `extName` is a string and lowercase.
```

Comment:
This lesson could probably be broken into two lessons, the first to make sure `extName` is lowercase and the second to ensure that it's a string.